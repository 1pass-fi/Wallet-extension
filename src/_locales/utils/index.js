export default async function getFetchWithTimeout(url, opts, timeout = 5000) {
  const abortController = new window.AbortController()
  const { signal } = abortController
  const f = window.fetch(url, {
    ...opts,
    signal
  })

  const timer = setTimeout(() => abortController.abort(), timeout)

  try {
    const res = await f
    clearTimeout(timer)
    return res
  } catch (e) {
    clearTimeout(timer)
    throw e
  }
}
