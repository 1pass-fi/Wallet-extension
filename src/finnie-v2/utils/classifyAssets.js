import filter from 'lodash/filter'

const classifyAssets = (assets, collections) => {
  try {
    // add collection name to nfts
    return assets.map(asset => {
      const collectionNames = collections.filter(collection => collection.collection?.includes(asset.txId)).map(collection => collection.name)

      asset.collection = collectionNames ? collectionNames : []
      return asset
    })
  } catch (error) {
    console.log('Fail to classify assets: ', error)
    return assets
  }
}

export default classifyAssets