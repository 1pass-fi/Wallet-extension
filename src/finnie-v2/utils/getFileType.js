import get from 'lodash/get'

export const NFT_TYPES = { IMAGE: 'image', VIDEO: 'video', AUDIO: 'audio', HTML: 'html' }

export const getFileType = (file) => {
  const fileType = get(file, 'type', '')
  for (let i in NFT_TYPES) {
    if (fileType.includes(NFT_TYPES[i])) {
      return NFT_TYPES[i]
    }
  }
  return ''
}
