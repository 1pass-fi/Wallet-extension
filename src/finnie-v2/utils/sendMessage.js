import { store } from 'react-notifications-component'

const SUCCESS = 'success'
const DANGER = 'danger'

const sendMessage = (type) => ({ title, message }) => {
  store.addNotification({
    title,
    message,
    type,
    insert: 'top',
    container: 'top-right',
    animationIn: ['animate__animated', 'animate__fadeIn'],
    animationOut: ['animate__animated', 'animate__fadeOut'],
    dismiss: {
      duration: 5000,
      onScreen: true
    }
  })
}

export default {
  success: sendMessage(SUCCESS),
  danger: sendMessage(DANGER)
}
