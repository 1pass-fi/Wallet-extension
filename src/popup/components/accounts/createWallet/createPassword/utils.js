import get from 'lodash/get'
import { ERROR_MESSAGE } from 'koiConstants'

export const validatePassword = ({ e, setError, generateWallet }) => {
  try {
    const pwd = get(e, 'target.pwd.value')
    const pwdConfirm = get(e, 'target.pwdConfirm.value')
    const checked = get(e, 'target.checkbox.checked')
    if (pwd.length < 8) {
      setError(ERROR_MESSAGE.PASSWORD_LENGTH)
    } else if (pwd !== pwdConfirm) {
      setError(ERROR_MESSAGE.PASSWORD_MATCH)
    } else if (!checked) {
      setError(ERROR_MESSAGE.CHECKED_TERMS)
    } else {
      generateWallet({ stage: 2, password: pwd })
    }
  } catch (err) {
    setError(err.message)
  }
}
