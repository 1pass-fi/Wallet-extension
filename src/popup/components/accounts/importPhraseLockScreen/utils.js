import { PATH, ERROR_MESSAGE } from 'constants'

export const validatePhrase = ({
  phrase,
  password,
  confirmPassword,
  history,
  setError,
  importWallet,
}) => {
  try {
    if (phrase.split(' ').length != 12) {
      setError(ERROR_MESSAGE.INCORRECT_PHRASE)
    } else if (password.length < 8) {
      setError(ERROR_MESSAGE.PASSWORD_LENGTH)
    } else if (password !== confirmPassword) {
      setError(ERROR_MESSAGE.PASSWORD_MATCH)
    } else {
      const redirectPath = PATH.IMPORT_KEY_REDIRECT
      importWallet({ data: phrase, password, history, redirectPath })
    }
  } catch (err) {
    /* istanbul ignore next */
    setError(err.message)
  }
}
