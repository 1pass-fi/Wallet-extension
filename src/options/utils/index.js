import { Web } from '@_koi/sdk/web'
import { MOCK_COLLECTIONS_STORE, STORAGE } from 'constants/koiConstants'
import { isString } from 'lodash'
import { find, get } from 'lodash'
import numeral from 'numeral'
import { getChromeStorage, setChromeStorage } from 'utils'

export const koi = new Web()

import storage from 'services/storage'

export const formatNumber = (value, decimal) => {
  const zeroArray = [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]
  return numeral(value).format(
    decimal ? `0,0.${zeroArray.slice(0, decimal).join('')}` : '0,0.00'
  )
}

export const getDisplayAddress = (address, head = 6, tail = 4) => {
  try {
    if (isString(address)) return `${address.slice(0, head)}...${address.slice(address.length - tail)}`
  } catch (err) {
    throw new Error(err.message)
  }
}

/**
 * 
 * @param {File} file file object
 */
export const saveImageDataToStorage = async (file) => {
  const url = URL.createObjectURL(file)
  // console.log('bottomButton- file', file)

  // get arrayBuffer
  const response = await fetch(url)
  const blob = await response.blob()
  const dataBuffer = await blob.arrayBuffer()
  // console.log('bottomButton- dataBuffer', dataBuffer)

  // create a 8bit array and save to local storage
  let u8 = new Int8Array(dataBuffer)
  // console.log('bottomButton- u8', u8)

  // save u8 to local storage
  u8 = JSON.stringify(u8, null, 2)
  // console.log('bottomButton- u8', u8)
  await setChromeStorage({ NFT_BIT_DATA: u8 })
  await storage.generic.set.nftBitData(u8)
}

/**
 *
 * @param {Array} collection { tx: [id1, id2] } 
 */
export const mockSaveCollections = async (collection) => {
  console.log('new collection: ', collection)
  const storage = await getChromeStorage(MOCK_COLLECTIONS_STORE)
  const collections = storage[MOCK_COLLECTIONS_STORE] || []
  collections.push(collection)

  await setChromeStorage({[MOCK_COLLECTIONS_STORE]: collections})
}

export const mockGetCollections = async () => {
  const storage = await getChromeStorage(MOCK_COLLECTIONS_STORE)
  const collections = storage[MOCK_COLLECTIONS_STORE] || []
  return collections
}

export const stringTruncate = (str, length) => {
  try {
    if (str.length > 20) {
      return `${str.slice(0,length)}...`
    }
    return str
  } catch (err) {
    console.log(err)
  }
}

export const JSONFileToObject = async (file) => {
  try {
    const fileText = await file.text()
    return JSON.parse(fileText)
  } catch (err) {
    throw new Error(err.message)
  }
}
