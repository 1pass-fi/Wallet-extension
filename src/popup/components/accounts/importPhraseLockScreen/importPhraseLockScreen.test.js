import '@babel/polyfill'
import { fireEvent, getByText, render } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import React from 'react'
import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store'
import { BrowserRouter as Router } from 'react-router-dom'

import ImportPhraseLockScreen from '.'

jest.mock('./utils')
import * as utils from './utils'

const mockStore = configureStore([])

describe('Test for ImportPhraseLockScreen component', () => {
  let setErrorMock, importWalletMock, storeMock, container
  beforeEach(() => {
    setErrorMock = jest.fn()
    importWalletMock = jest.fn()

    const initState = {
      setError: setErrorMock,
      importWallet: importWalletMock,
    }

    storeMock = mockStore(initState)

    container = render(
      <Router>
        <Provider store={storeMock}>
          <ImportPhraseLockScreen />
        </Provider>
      </Router>
    ).container
  })

  describe('renders without crashing', () => {
    it('renders correctly', () => {
      expect(container).toMatchSnapshot()
    })
  })

  describe('onSubmit', () => {
    it('call validatePhrase', () => {
      const paswordField = document.querySelector('.password .input')
      const confirmPaswordField = document.querySelector(
        '.confirm-password .input'
      )
      const phraseField = document.querySelector('.seed-phrase .input')
      const submitButton = document.querySelector('.unlock-button')

      fireEvent.change(paswordField, { target: { value: '12345678' } })
      fireEvent.change(confirmPaswordField, { target: { value: '12345678' } })
      fireEvent.change(phraseField, {
        target: { value: '1 2 3 4 5 6 7 8 9 0 1 2' },
      })
      fireEvent.click(submitButton)

      expect(utils.validatePhrase).toBeCalled()
    })
  })
})
