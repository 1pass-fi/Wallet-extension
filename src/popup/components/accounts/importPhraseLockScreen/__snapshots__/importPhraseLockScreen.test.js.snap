// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for ImportPhraseLockScreen component renders without crashing renders correctly 1`] = `
<div>
  <div
    class="seed-phrase-lock-screen"
  >
    <header
      class="lock-screen-header"
    >
      <svg
        class="koi-icon"
      />
      <div
        class="koi-title"
      >
        Koi Wallet
      </div>
    </header>
    <div
      class="content"
    >
      <div
        class="welcome"
      >
        Welcome
      </div>
      <div
        class="text-area-field seed-phrase field"
      >
        <div
          class="label"
        >
          Paste your seed phrase
        </div>
        <div
          class="input-wrapper"
        >
          <textarea
            class="input protect"
            name="inputPhrase"
            placeholder=""
            spellcheck="false"
          />
          <svg
            class="toggle-display"
          />
        </div>
      </div>
      <div
        class="input-field password field"
      >
        <div
          class="label"
        >
          password
        </div>
        <div
          class="input-wrapper"
        >
          <input
            class="input"
            name="password"
            placeholder="Make it unique (min. 8 characters)"
            type="password"
            value=""
          />
          <svg
            class="toggle-display"
          />
        </div>
      </div>
      <div
        class="input-field confirm-password field"
      >
        <div
          class="label"
        >
          Confirm password
        </div>
        <div
          class="input-wrapper"
        >
          <input
            class="input"
            name="confirm-password"
            placeholder=""
            type="password"
            value=""
          />
          <svg
            class="toggle-display"
          />
        </div>
      </div>
      <button
        class="button-shared button-filled unlock-button"
      >
        Unlock
      </button>
      <div
        class="login-with-password"
      >
        Log in with my
         
        <a
          class="password-link"
          href="/account/login"
        >
          password
        </a>
      </div>
    </div>
  </div>
</div>
`;
